src/pages/address/[principal].tsx

```diff
@@ -111,52 +112,54 @@ const AddressPage: NextPage<any> = ({ error }) => {
 };
 
 const prefetchData = async (
-  networkUrl: string,
   addressPageQuery: string,
-  res: ServerResponse
+  res: ServerResponse,
+  networkUrl?: string
 ): Promise<QueryClient> => {
   const queryClient = new QueryClient();
-  const prefetchOptions = { staleTime: 5000 };
-  const queries = getAddressQueries(networkUrl);
-  try {
-    await Promise.all([
-      queryClient.prefetchQuery(
-        addressQK(AddressQueryKeys.coreApiInfo),
-        queries.fetchCoreApiInfo(),
-        prefetchOptions
-      ),
-      queryClient.prefetchQuery(
-        addressQK(AddressQueryKeys.accountInfo, addressPageQuery),
-        queries.fetchAccountInfo(addressPageQuery),
-        prefetchOptions
-      ),
-      queryClient.prefetchInfiniteQuery(
-        addressQK(AddressQueryKeys.mempoolTransactionsForAddress, addressPageQuery),
-        queries.fetchMempoolTransactionsForAddress(addressPageQuery),
-        prefetchOptions
-      ),
-      queryClient.prefetchInfiniteQuery(
-        addressQK(AddressQueryKeys.transactionsForAddress, addressPageQuery),
-        queries.fetchTransactionsForAddress(addressPageQuery),
-        prefetchOptions
-      ),
-      queryClient.prefetchQuery(
-        addressQK(AddressQueryKeys.accountBalance, addressPageQuery),
-        queries.fetchAccountBalance(addressPageQuery),
-        prefetchOptions
-      ),
-    ]);
-  } catch (err) {
-    res.statusCode = err.status;
+  if (networkUrl) {
```

Comment:
I often prefer using guard clause to keep the main code logic flat. I think it's a bit easier to read and prevents too deeply nested logic.
```javascript
if (!networkUrl) return
const prefetchOptions = { staleTime: 5000 };
const queries = getAddressQueries(networkUrl);
try {
...
```
Same comment below